### PR TITLE
Reusable code libs 62

### PR DESCRIPTION
**Pull request by kmahmood74**
see detailed description in https://github.com/EnsembleUI/ensemble/issues/62

### CODE REVIEW

**Comment by kmahmood74** [comment]
see detailed description in #62

File: lib/framework/scope.dart [hunk 1]
```diff
@@ -151,11 +152,13 @@ class ScopeManager extends IsScopeManager with ViewBuilder, PageBindingManager {
...
   }
 
-  /// create a copy of the parent's data scope
+  /// create a copy of the parent's data scope. the initialMap is generally coming from the imports
   @override
-  ScopeManager createChildScope({bool ephemeral = false}) {
+  ScopeManager createChildScope({bool ephemeral = false, List<ParsedCode>? importedCode}) {
     ScopeManager childScope =
-        ScopeManager(dataContext.clone(), pageData, ephemeral: ephemeral);
+        ScopeManager(dataContext
```
Comment (vusters) [inline lib/framework/scope.dart]: So evalImports makes sense at the screen level as we want the screens to have separate context, but within a screen I would want my widgets, apis, item-templates,... to use the same code context. This is how the screen's Code block works today. createChildScope() is used for these purposes.

Comment (kmahmood74) [reply lib/framework/scope.dart]: I am not following you. A re-usable custom widget may have its own imports. So when we evaluate custom widget, we evaluate its imports as well.

Comment (vusters) [reply lib/framework/scope.dart]: talked to Khurram. This is not what I understood. All good

**Review by vusters — APPROVED** [review]

**Pull request merged**